src/test/java/net/sf/jabref/importer/ImportFileFilterTest.java

```diff
@@ -0,0 +1,92 @@
+package net.sf.jabref.importer;
+
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import net.sf.jabref.importer.fileformat.ImportFormat;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class ImportFileFilterTest {
+
+    private ImportFormat importFormatMultipleExtensions;
+    private ImportFormat importFormatSingleExtension;
+
+    @Before
+    public void setUp() {
+        importFormatSingleExtension = new ImportFormat() {
+            @Override
+            protected boolean isRecognizedFormat(BufferedReader input) throws IOException {
+                return false;
+            }
+
+            @Override
+            protected ParserResult importDatabase(BufferedReader input) throws IOException {
+                return null;
+            }
+
+            @Override
+            public String getFormatName() {
+                return "Single Extension";
+            }
+
+            @Override
+            public List<String> getExtensions() {
+                return Collections.singletonList(".abc");
+            }
+
+            @Override
+            public String getDescription() {
+                return null;
+            }
+        };
+
+        importFormatMultipleExtensions = new ImportFormat() {
+            @Override
+            protected boolean isRecognizedFormat(BufferedReader input) throws IOException {
+                return false;
+            }
+
+            @Override
+            protected ParserResult importDatabase(BufferedReader input) throws IOException {
+                return null;
+            }
+
+            @Override
+            public String getFormatName() {
+                return "Multiple Extensions";
+            }
+
+            @Override
+            public List<String> getExtensions() {
+                return Arrays.asList(".abc", ".xyz");
+            }
+
+            @Override
+            public String getDescription() {
+                return null;
+            }
+        };
```

Comment:
I would create these mocks in the corresponding test. As it is always a little bit annoying if the before part initializes a lot of different things, but the tests use always their own subset. In that case, you have a 1-1 relation between test and your mock classes.